packages/block-editor/src/components/block-list/block-list-item.native.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { ReadableContentView } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import BlockListBlock from './block';
+import BlockInsertionPoint from './insertion-point';
+
+const stretchStyle = {
+	flex: 1,
+};
+
+export class BlockListItem extends Component {
+	render() {
+		const {
+			clientId,
+			isReadOnly,
+			shouldShowInsertionPointBefore,
+			shouldShowInsertionPointAfter,
+			contentResizeMode,
+			shouldShowInnerBlockAppender,
+			...restProps
+		} = this.props;
+		const readableContentViewStyle =
+			contentResizeMode === 'stretch' && stretchStyle;
+		return (
+			<ReadableContentView style={ readableContentViewStyle }>
+				<View
+					style={ readableContentViewStyle }
+					pointerEvents={ isReadOnly ? 'box-only' : 'auto' }
+				>
+					{ shouldShowInsertionPointBefore && (
+						<BlockInsertionPoint />
+					) }
+					<BlockListBlock
+						key={ clientId }
+						showTitle={ false }
+						clientId={ clientId }
+						{ ...restProps }
+					/>
+					{ ! shouldShowInnerBlockAppender() &&
+						shouldShowInsertionPointAfter && (
+							<BlockInsertionPoint />
+						) }
+				</View>
+			</ReadableContentView>
+		);
+	}
+}
+
+export default compose( [
+	withSelect(
+		( select, { rootClientId, isStackedHorizontally, clientId } ) => {
+			const {
+				getBlockOrder,
+				getBlockInsertionPoint,
+				isBlockInsertionPointVisible,
+				getSettings,
+			} = select( 'core/block-editor' );
+
+			const blockClientIds = getBlockOrder( rootClientId );
+			const insertionPoint = getBlockInsertionPoint();
+			const blockInsertionPointIsVisible = isBlockInsertionPointVisible();
+			const shouldShowInsertionPointBefore =
+				! isStackedHorizontally &&
+				blockInsertionPointIsVisible &&
+				insertionPoint.rootClientId === rootClientId &&
+				// if list is empty, show the insertion point (via the default appender)
+				( blockClientIds.length === 0 ||
+					// or if the insertion point is right before the denoted block
+					blockClientIds[ insertionPoint.index ] === clientId );
+
+			const shouldShowInsertionPointAfter =
```

Comment:
I know this `shouldShowInsertionPointAfter` name was being used before and we're just continuing to use it here, but I feel like it may be good to update this a name to reflect that it is only checking if the insertion point should go at the end. The way the current `shoudlShowInsertionPointBefore` and `shouldShowInsertionPointAfter` names echo each other makes it feel like they have similar behavior and only differ regarding whether the insertion point is shown before or after a given block, which makes it confusing (well, I was confused) when you look and see that `shouldShowInsertionPointAfter` is actually only checking if the insertion point should be shown at the very end of the list. 

I might just be misunderstanding how this is being used though. WDYT?

Reply:
I have mixed feelings about this one :) That's right we show the insertion point after the last block in inner blocks, and yes, the logic is a bit different than in case of `shoudlShowInsertionPointBefore` but it still says whether we should render insertion point after that block or not. I'm open to suggestions :) Btw. Iit was not the main goal of that PR but I agree that it could be refactored here

Reply:
Fair enough. I certainly don't feel like this is a blocker, so I'm fine leaving it unchanged if you prefer.